bifrost_min_read_check/launcher.py

```diff
@@ -3,139 +3,169 @@
 Launcher file for accessing dockerfile commands
 """
 import argparse
+from io import FileIO, StringIO
 import json
-import subprocess
 import os
 import sys
 import traceback
+import subprocess
 from bifrostlib import datahandling
+from bifrostlib.datahandling import Component
+from bifrostlib.datahandling import ComponentReference
+import yaml
+import pprint
+from typing import List, Dict
+
+from yaml.tokens import StreamEndToken
+
+
+global COMPONENT
+
+
+def initialize():
+    with open(os.path.join(os.path.dirname(__file__), 'config.yaml')) as fh:
+        config: Dict = yaml.load(fh, Loader=yaml.FullLoader)
+
+    if not(datahandling.has_a_database_connection()):
+        raise ConnectionError("BIFROST_DB_KEY is not set or other connection error")
+
+    global COMPONENT
+    try:
+        component_ref = ComponentReference(name=config["name"])
+        COMPONENT = Component.load(component_ref)
+        if COMPONENT is not None and '_id' in COMPONENT.json:
+                return
+        else:
+            COMPONENT = Component(value=config)
+            install_component()
+
+    except Exception as e:
+        print(traceback.format_exc(), file=sys.stderr)
+    return
+
+
+def install_component():
+    COMPONENT['install']['path'] = os.path.os.getcwd()
+    print(f"Installing with path:{COMPONENT['install']['path']}")
+    try:
+        COMPONENT.save()
+        print(f"Done installing")
+    except:
+        print(traceback.format_exc(), file=sys.stderr)
+        sys.exit(0)
+
 
-COMPONENT: dict = datahandling.load_yaml(os.path.join(os.path.dirname(__file__), 'config.yaml'))
+class types():
```

Comment:
Is this used? if its a class it should be named Types, but I don't see it used (in this file at least)